src/compute/generic.ts

```diff
@@ -7,14 +7,29 @@ import { Filters, generateFiltersQuery } from '../filters'
 import { ratioToPercentage } from './common'
 import { getEntity } from '../helpers'
 import { getParticipationByYearMap } from './demographics'
+import { useCache } from '../caching'
 
-interface TermAggregationByYearOptions {
+const firstYear = 2016
+const currentYear = new Date().getFullYear()
+const allYears = _.range(firstYear, currentYear)
```

Comment:
I think this is not really flexible, it depends on the survey type, so we would end up making queries while we know we don't have any data, we could still keep automatic year detection based on the query, I'm confident there's a way to get around the `limit`.